modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/transaction/FeeTransactionValidator.scala

```diff
@@ -0,0 +1,71 @@
+package org.tessellation.node.shared.domain.transaction
+
+import cats.data.{NonEmptyList, ValidatedNec}
+import cats.effect.Async
+import cats.syntax.all._
+
+import org.tessellation.currency.dataApplication.FeeTransaction
+import org.tessellation.ext.cats.syntax.validated._
+import org.tessellation.node.shared.domain.transaction.FeeTransactionValidator.FeeTransactionValidationErrorOr
+import org.tessellation.schema.address.Address
+import org.tessellation.security.signature.SignedValidator.SignedValidationError
+import org.tessellation.security.signature.{Signed, SignedValidator}
+
+import derevo.cats.{eqv, show}
+import derevo.derive
+
+trait FeeTransactionValidator[F[_]] {
+  def validate(signedTransaction: Signed[FeeTransaction]): F[FeeTransactionValidationErrorOr[Signed[FeeTransaction]]]
+  def validateMultiple(
+    signedTransactions: NonEmptyList[Signed[FeeTransaction]]
+  ): F[FeeTransactionValidationErrorOr[NonEmptyList[Signed[FeeTransaction]]]]
+}
+
+object FeeTransactionValidator {
+  def make[F[_]: Async](
+    signedValidator: SignedValidator[F]
+  ): FeeTransactionValidator[F] =
+    new FeeTransactionValidator[F] {
+      def validate(
+        signedTransaction: Signed[FeeTransaction]
+      ): F[FeeTransactionValidationErrorOr[Signed[FeeTransaction]]] =
+        for {
+          srcAddressSignatureV <- validateSourceAddressSignature(signedTransaction)
+          differentSrcAndDstV = validateDifferentSourceAndDestinationAddress(signedTransaction)
+        } yield
+          srcAddressSignatureV
+            .productR(differentSrcAndDstV)
+
+      def validateMultiple(
+        signedTransactions: NonEmptyList[Signed[FeeTransaction]]
+      ): F[FeeTransactionValidationErrorOr[NonEmptyList[Signed[FeeTransaction]]]] =
```

Comment:
its name can be just `validate`. You'd see then two signatures:
```scala
validate(signedTransaction: Signed[FeeTransaction])
validate(signedTransactions: NonEmptyList[Signed[FeeTransaction]]
```

It's okay to overload arguments, so you can use a single name `validate` with different parameters. It does not make sense to create new names like `validateMultiple`

Reply:
👍🏼